mpt/core/src/main/java/org/apache/james/mpt/api/DomainAdder.java

```diff
@@ -0,0 +1,33 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mpt.api;
+
+/**
+ * Adds domains on demand.
+ */
+public interface DomainAdder {
+    
+    /**
+     * Adds a domain.
+     * @param domain not null
+     * @throws Exception when domain cannot be added
+     */
+    void addDomain(String domain) throws Exception;
```

Comment:
what is the expected behavior when a domain is already existing? if domain is empty string? is blank?

Reply:
or maybe we don't really want to write javadoc (:

Reply:
i don't think we need to go into much details now.
The interface is used only during test.
One of the two implementation is a Noop one.

Reply:
ok so drop the javadoc, we better not write doc if it's partial